updater/aws_test.go

```diff
@@ -15,6 +15,60 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetCommandResult(t *testing.T) {
+	cases := []struct {
+		name          string
+		invocationOut *ssm.GetCommandInvocationOutput
+		expectedError error
+		expectedOut   []byte
+	}{
+		{
+			name: "getCommand success",
+			invocationOut: &ssm.GetCommandInvocationOutput{
+				Status:                aws.String("Success"),
+				StandardOutputContent: aws.String("OutputContent"),
+			},
+			expectedOut:   []byte(aws.StringValue(aws.String("OutputContent"))),
+			expectedError: nil,
+		},
+		{
+			name:          "getCommand fail",
+			expectedError: fmt.Errorf("failed to retrieve command invocation output: failed to get command invocation"),
+			invocationOut: nil,
+			expectedOut:   nil,
+		},
+		{
+			name: "command status non-Success",
+			invocationOut: &ssm.GetCommandInvocationOutput{
+				Status:                aws.String("TimedOut"),
+				StatusDetails:         aws.String("Delivery Timed Out"),
+				StandardOutputContent: nil,
+			},
+			expectedError: fmt.Errorf("command %s has not reached success status, current status %q", "command-id", "Delivery Timed Out"),
+			expectedOut:   nil,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockSSM := MockSSM{
+				GetCommandInvocationFn: func(input *ssm.GetCommandInvocationInput) (*ssm.GetCommandInvocationOutput, error) {
+					assert.Equal(t, "command-id", aws.StringValue(input.CommandId))
+					assert.Equal(t, "instance-id", aws.StringValue(input.InstanceId))
+					return tc.invocationOut, tc.expectedError
+				},
+			}
+			u := updater{ssm: mockSSM}
+			actual, err := u.getCommandResult("command-id", "instance-id")
+			if tc.expectedOut != nil {
+				require.NoError(t, err)
+				assert.EqualValues(t, tc.expectedOut, actual)
+			} else {
+				require.Error(t, tc.expectedError, err)
```

Comment:
```suggestion
				require.Error(t, err)
				assert.Equal(t, tc.expectedError, err)
```

Reply:
Ended up having to adjust the tests around a little bit for this. As a result found I actually need `getError` in the `cases` struct after-all so put it back in.